### ISSUE TITLE
Command-Line Automation - Extract MKV Subtitles Tracks

### ISSUE THREAD

**Issue by KristjanLaane** [SubtitleEdit/subtitleedit#1754]
Would it be possible to have command line or some other scriptable method for exporting subtitles embedded into mkv? basically automating the import from mkv and then save operations which can be done manually with the GUI. Or is that possible already?
- Not sure what should happen if many subtitle tracks embedded in the matroska container?

**Comment by xylographe** [comment]
There is some support (not VobSub / PGS) in [CommandLineConvert](https://github.com/SubtitleEdit/subtitleedit/blob/master/src/Logic/CommandLineConvert.cs#L274). Run `SubtitleEdit /?` in your favourite shell for usage info.

AFAIK all supported subtitle tracks will be processed. Track number is used to generate different file names.

**Comment by KristjanLaane** [comment]
Brilliant!

This worked as an example:
`SubtitleEdit /convert "C:\Users\Hoover Dam\Downloads\InterestingDocumentary.mkv" SubRip`

Although one of the tracks failed, because I ran out of memory on the computer while the SubtitleEdit process used more than 3GB of RAM, growing steadily to that as it processed each track (the mkv file itself was about 1.5GB - not sure whether that's relevant):
`1: InterestingDocumentary_2_.mkv -> C:\Users\Hoover Dam\Downloads\InterestingDocumentary_2_.srt...Exception of type 'System.OutOfMemoryException' was thrown.`

p.s. SubtitleEdit /? did not show usage info, but launched the GUI, whereas the following worked :) SubtitleEdit /convert /?

**Comment by xylographe** [comment]
> SubtitleEdit /? did not show usage info

Sorry, my mistake, `SubtitleEdit /?` only works in SE beta version.

Running out of memory shouldn't happen. Tracks are processed sequentially.
How many subtitle tracks (only supported formats) does this mkv contain?

**Comment by KristjanLaane** [comment]
Ok, did some more testing and strange things are happening. I had a look at InterestingDocumentary.mkv and it actually has 1 subtitle track inside the container (as shown by VLC Media Player), whereas if I try to export it through command-line SubtitleEdit then it thinks there are 3. So I tried another mkv file, same problem, as well as massive RAM usage. And I tried a third mkv file, which also has 1 subtitle track and also produced output along the same lines:

```
t:\appsets\SubtitleEdit>SubtitleEdit /convert "C:\adhoc\OneSubtitleTrack.mkv" SubRip

t:\appsets\SubtitleEdit>

Subtitle Edit 3.4.12 - Batch converter

- Syntax: SubtitleEdit /convert <pattern> <name-of-format-without-spaces> [/offset:hh:mm:ss:ms] [/encoding:<encoding name>] [/fps:<frame rate>] [/targetfps:<frame rate>] [/inputfolder:<input folder>] [/outputfolder:<output folder>] [/removetextforhi] [/fixcommonerrors] [/pac-codepage:<code page>]

    example: SubtitleEdit /convert *.srt sami
    list available formats: SubtitleEdit /convert /list

1: OneSubtitleTrack_1_.mkv -> C:\adhoc\OneSubtitleTrack_1_.srt... done.
1: OneSubtitleTrack_2_.mkv -> C:\adhoc\OneSubtitleTrack_2_.srt... done.
1: OneSubtitleTrack_3_.mkv -> C:\adhoc\OneSubtitleTrack_3_.srt... done.

3 file(s) converted

```

And before running this I closed most of my other applications, because the SubtitleEdit process used up more than 3GB of RAM and if I hadn't closed my other applications it would have resulted in OutOfMemoryException again. 

Furthermore, the .srt files that are generated are 800-900MB in size each! :D (screenshot attached)
![hugesrtfiles](https://cloud.githubusercontent.com/assets/1708294/15276079/d57cb88a-1ae5-11e6-8e16-bedda3c07f00.PNG)

However, if I open SubtitleEdit GUI and open that very same mkv then it sees one subtitle track and imports it without a problem and without massive RAM usage. And that happens in less than a minute through the GUI whereas the command-line took 5 to 10 minutes. 
Clearly, something is wrong with the command-line implementation when the same functionality works through the GUI. 

I have uploaded the OneSubtitleTrack.mkv file for testing it on your side, it is about 500MB:
https://www.dropbox.com/s/32r1ize57pblth5/OneSubtitleTrack.mkv?dl=0
(use Download button on top right)

Hope you can figure out what is going on?

**Comment by xylographe** [comment]
Thank you!

Fixed in PR #1758.

**Comment by KristjanLaane** [comment]
tried SubtitleEdit-3.4.12.252-issue1754.7z and a world of difference, no memory problems, very fast and the correct subtitle file appeared:

```
t:\appsets\SubtitleEdit>SubtitleEdit /convert "C:\adhoc\OneSubtitleTrack.mkv" SubRip

t:\appsets\SubtitleEdit>
Subtitle Edit 3.4.12 - Batch converter

1: OneSubtitleTrack.mkv -> C:\adhoc\OneSubtitleTrack.srt... done.

1 file(s) converted

```

thanks a great deal!
